### PR TITLE
RUST-1228 Create a skeleton for a mdbook manual

### PR DESCRIPTION
**Pull request by abr-egn**
Happily, [mdbook](https://rust-lang.github.io/mdBook/index.html) is quite straightforward.

### CODE REVIEW

**Review by kmahar — APPROVED** [review]
LGTM - have you thought about how we host/serve this? if GH pages, we should think about whether we want to use a separate `gh-pages` branch for the book

**Comment by abr-egn** [comment]
> LGTM - have you thought about how we host/serve this? if GH pages, we should think about whether we want to use a separate `gh-pages` branch for the book

From our in-person discussion, GH pages seems like the clear way to go, and at least initially we won't be using a separate branch.  I've updated this PR to point the output of mdbook at `docs/manual` and include the generated files.  I also added a `.gitattributes` file so the contents of that are marked as generated, which should mean they don't count towards language stats for the repo.  Unfortunately, I couldn't find any way to exclude it from the contribution graph.

File: manual/src/SUMMARY.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+- [Introduction](README.md)
+
+# Basic Usage
+
+- [Installation](basic/installation.md)
+- [Feature Choices](basic/feature_choices.md)
+- [Connection and Operation](./basic/connection_and_operation.md)
+
+# Advanced
```
Comment (patrickfreed) [inline manual/src/SUMMARY.md:11]: IMO we can drop the "basic" vs "advanced" distinction. I'm a bit worried that users may potentially skip over the advanced stuff when getting started, even though ideally all users, from beginners to experts, should be using serde, for example.

File: manual/src/SUMMARY.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
```
Comment (patrickfreed) [inline manual/src/SUMMARY.md:12]: I think some more good sections to have would be "Cursors", "Sessions and Transactions", "Change Streams", and "Monitoring" (for stuff like command, CMAP, and SDAM monitoring). Perhaps, we'd want "Reading from the Database" and "Writing to the Database" sections too, and "Cursors" could just be a subsection of the former.

**Review by patrickfreed — COMMENTED** [review]
format looks great! I just have some comments on the proposed section breakdown.

File: manual/src/SUMMARY.md (hunk 1 above)
Comment (abr-egn) [reply manual/src/SUMMARY.md]: Done.

File: manual/src/SUMMARY.md (hunk 2 above)
Comment (abr-egn) [reply manual/src/SUMMARY.md]: Done.

**Review by patrickfreed — APPROVED** [review]

**Review by isabelatkinson — APPROVED** [review]

**Pull request merged**